pkg/controllers/cassandra/service/control.go

```diff
@@ -34,9 +37,17 @@ func NewControl(
 
 func (e *defaultCassandraClusterServiceControl) Sync(cluster *v1alpha1.CassandraCluster) error {
 	svc := ServiceForCluster(cluster)
-	_, err := e.kubeClient.CoreV1().Services(svc.Namespace).Update(svc)
+	client := e.kubeClient.CoreV1().Services(svc.Namespace)
+	existingSvc, err := e.serviceLister.Services(cluster.Namespace).Get(svc.Name)
```

Comment:
(nit, but could be significant in future if we ever mess with namespaces - `svc.Namespace` or `cluster.Namespace`? I'd say `svc.Namespace` should be used, as the `ServiceForCluster` function takes care of ensuring that is correct.)

Reply:
Good point. Done.